[Tang Zhiying] iP

zhing22 opened the pull request:
# TiTi Catbot

> <h3>"Meow Meow Meowww~ ฅ^•ﻌ•^ฅ" – TiTi </h3> 
> <h6> (<strike>and yes whatever TiTi says shall be correct</strike>)</h5> 

<br> 

TiTi Catbot frees your mind of having to remember things you need to do. It's, 
* text-based
* SUPER easy to learn
* SUPER SUPER fast to use (, and most importantly)
* SUPER SUPER SUPER CUTE!

All you need to do is, 
1. download it from [here](https://github.com/zhing22/ip).
2. and start chatting with TiTi! 🐾

And it is __FREE__!!!

<hr> 

Features: 
- [X] Managing tasks
- [x] Managing deadlines and events
- [ ] Reminders (coming soon)
- [ ] Cute pictures of TiTi (coming soon)

<hr>

If you are programmer Ling Ling, you can use it to practice Java 40 hours a day too. Here's the ```main``` method: 

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

zhing22 committed:
Level-1

zhing22 committed:
Level-2: : Add, List

zhing22 committed:
Level-3: Mark as Done

zhing22 committed:
Level-4: ToDo, Event, Deadline

zhing22 committed:
A-TextUiTesting

zhing22 committed:
Level-5: Handle Errors

zhing22 committed:
Level-6: Delete

zhing22 committed:
Add function to save history in hard disk

zhing22 committed:
Add ability to understand dates and time

zhing22 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add ability to understand dates and time

zhing22 committed:
Use more OOP

Extracted following classes:
Ui: deals with interactions with the user
Parser: deals with making sense of user command
TaskList: contains the task list

zhing22 committed:
Divide classes into packages

Added package duke.util and duke.task

zhing22 committed:
Add JavaDocs

zhing22 committed:
Slight tweak to align with coding standard

zhing22 committed:
Add function to search task using keyword

zhing22 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Slight tweak to align with coding standard

# Conflicts:
#	src/main/java/TiTi/util/Parser.java
#	src/main/java/TiTi/util/TaskList.java
#	src/main/java/TiTi/util/Ui.java

zhing22 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add function to search task using keyword

# Conflicts:
#	src/main/java/TiTi/util/Response.java
#	src/main/java/TiTi/util/TaskList.java

zhing22 committed:
Update README using GFMD syntax

zhing22 committed:
Slight changes to README

zhing22 committed:
Add JUnit Test to Deadline and TaskList class

tanqiya commented on src/main/java/TiTi/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+        checkDate(by);
+    }
+
+    private void checkDate(String by) {
```
Perhaps this function could be named with a verb, such as `dateChecker`.

zhing22 replied:
thanks for the review! but ```dateChecker``` is actually a noun! And by convention class and variable should be names with nouns, and methods should be named with verb! Just saying in case you might have made the same assumptions for your own code ;-;

tanqiya commented on src/main/java/TiTi/util/Parser.java:
```diff
@@ -0,0 +1,124 @@
+        String cue = input.split(" ", 2)[0];
+
+        // handle missing description
+        if ((cue.equals("todo") || cue.equals("deadline")
+                || cue.equals("event") || cue.equals("find"))
+                && input.split(" ", 2).length == 1) {
```
This expression seems a little bit complicated. Perhaps you could set aside a boolean `isDescriptionMissing` to handle this checking.

tanqiya commented on src/main/java/TiTi/util/Ui.java:
```diff
@@ -0,0 +1,137 @@
+ * Prints to the user interface.
+ */
+public class Ui {
+    private static final String STARTER_NORMAL =   "   (=^ ･ｪ･^=) < ";
+    private static final String STARTER_BUFFER =   "                ";
+    private static final String STARTER_SLEEPY =   "   (=^ ‐ｪ‐^=) < ";
+    private static final String STARTER_CONFUSED = "   (=^ ･x･^=)? < ";
+    private static final String STARTER_HAPPY =    "   (=^ ･ω･^=)❀ < ";
```
so cute!! 😸

tanqiya commented on src/main/java/TiTi/task/ToDo.java:
```diff
@@ -0,0 +1,23 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Do remember to add a new line here at the end of each file.

tanqiya commented on src/main/java/TiTi/util/Parser.java:
```diff
@@ -0,0 +1,124 @@
+
+        case "done":
+            taskNumber = Integer.parseInt("" + input.charAt(5));
+            if (taskNumber > numberOfTasks) {
+                return new Response(Response.Cue.TASKERROR);
+            } else {
+                Task task = taskList.get(taskNumber - 1);
+                task.complete();
+                return new Response(Response.Cue.DONE, task);
+            }
```
```suggestion
            if (taskNumber > numberOfTasks) {
                return new Response(Response.Cue.TASKERROR);
            }
            Task task = taskList.get(taskNumber - 1);
            task.complete();
            return new Response(Response.Cue.DONE, task);
            
```
Since a new `Response` is returned in the if-block, the else-block is not required and can be removed so the main path is un-indented.
I noticed the same issue in several other places too.

xyliew25 commented on src/main/java/TiTi.java:
```diff
@@ -0,0 +1,36 @@
+import TiTi.util.SavedHistory;
```
According to the coding standard, every class should be part of some package. Perhaps `package titi;`

xyliew25 commented on src/main/java/TiTi/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+package TiTi.task;
```
According to the coding standard, names representing packages should be in all lower case. So in this case, perhaps `package titi.task;`? This applies to the other files as well.

xyliew25 commented on src/main/java/TiTi/util/Parser.java:
```diff
@@ -0,0 +1,124 @@
+import TiTi.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
```
Not sure if there is a fixed import order convention, but I think as long as it is consistent throughout the project, it is alright. According to this [link](https://se-education.org/guides/tutorials/intellijCodeStyle.html), the above import is reversed.

xyliew25 commented on src/main/java/TiTi/util/Response.java:
```diff
@@ -0,0 +1,36 @@
+ * Contain the task to be printed (if needed).
+ */
+public class Response {
+    public enum Cue {EXIT, LIST, DONE, DELETE, TASKERROR, TODO, DEADLINE,
```
Not sure if enum names should have underscore to separate two words, as in like constants.

xyliew25 commented on src/main/java/TiTi/util/TaskList.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return size of list
+     */
+    public int size() {
```
According to the coding standards, names representing methods must be verbs. So perhaps `getSzie`?

xyliew25 commented on src/main/java/TiTi/util/Ui.java:
```diff
@@ -0,0 +1,137 @@
+    /**
+     * Prints the response for the user command onto the user interface.
+     */
+    public void userCommand() {
```
Similarly, perhaps `printUserCommand`?

xyliew25 commented on src/test/java/TiTi/task/DeadlineTest.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    @Test
+    public void DateTest(){
```
According to coding standards, names representing methods must be verbs and written in camelCase. So perhaps `testDate()`?

xyliew25 commented on src/test/java/TiTi/task/DeadlineTest.java:
```diff
@@ -0,0 +1,22 @@
+package TiTi.task;
+
+import TiTi.task.Deadline;
+import org.junit.jupiter.api.Test;
```
Regardless of the import order, I think there should be a new line separate these two imports as they are from different sources.

xyliew25 commented on src/main/java/TiTi/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+
+/**
+ * Represents the Deadline task.
+ * Contain String description of the task.
+ * Contain String description of the date of task.
+ * Contain LocalDate object of the date of task (if applicable).
+ * Contain boolean value of whether the task has been completed.
```
Should be `Contains` instead of `Contain`. These are probably overlooked but I noticed the same issue in several other places too.

xyliew25 commented on src/main/java/TiTi/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+    protected LocalDate date;
+
+    /**
+     * Constructor for Deadline class.
```
Not sure if nouns are allowed. If not, perhaps can put `Constructs a Deadline instance.` ?

xyliew25 reviewed (commented):
There are quite a few coding standards violations, but kudos to all the effort you put into your iP! It is indeed a CUTEE chatbot!! Also, do correct me if I misunderstood or commented something wrongly :)

zhing22 committed:
Add Gradle

zhing22 committed:
Testing out GUI

zhing22 committed:
Add initial GUI

zhing22 committed:
Finalise Gui

zhing22 committed:
Generate new JAR file using Gradle

Package the app as an executable JAR file so that it can be distributed easily.

zhing22 committed:
Create gradle.yml

zhing22 committed:
Add assertion

Input description from user when adding tasks might be empty. If so, response would be taken care in UI response. Hence tasks with empty description should not be created.

* Assertion test is used to stop program if attempt to initialise empty description tasks

zhing22 committed:
Improve code quality.

Examines the code and refactor to improve the code quality where necessary:
* break long method into smaller sub-methods
* improve variable naming
* improve documentation style
* other slight changes

zhing22 committed:
Merge pull request #1 from zhing22/branch-A-CodeQuality

Improve code quality

zhing22 committed:
Merge branch 'branch-A-CodeQuality' into branch-A-Assertions

* branch-A-CodeQuality:
  Improve code quality.

zhing22 committed:
Add mass operations functionality.

Provide a way to perform tasks on multiple items:
* Multiple tasks can be deleted at once
* Multiple tasks can be marked done at once

zhing22 committed:
Update GUI

zhing22 committed:
Set theme jekyll-theme-merlot

zhing22 committed:
Update code quality

changes to:
*package naming
*fix header comments for TiTi.main()

zhing22 committed:
Merge branch 'master' of https://github.com/zhing22/ip

* 'master' of https://github.com/zhing22/ip:
  Set theme jekyll-theme-merlot

zhing22 committed:
Add User Guide

Includes a quick description of TiTi Chatbot, and a list of all usages and commands

zhing22 committed:
User Guide test

zhing22 committed:
Finalise user guide

zhing22 committed:
Finalise user guide

zhing22 committed:
Finalise User Guide

zhing22 committed:
Fix code standard violations

zhing22 committed:
Fix multiple bugs

* Wrong picture namings
* Problem with running program
* Problems with delete and done functionalites

zhing22 committed:
Fix typo in User Guide

zhing22 committed:
Update README.md
